### DISCUSSION TITLE
Using AI tools for MyTourbook Development

### DISCUSSION THREAD

**Discussion by FJBDev** [mytourbook/mytourbook#1328] [General]
I've been trying to use [this plugin](https://github.com/orgs/community/discussions/51197#discussioncomment-7684246) when working on MyTourbook development as I find AI very helpful to speed up programming. I have also previewed [the TabNine plugin](https://marketplace.eclipse.org/content/tabnine-ai-assistant-chat-software-developers) that showed good promises

When I go [here](https://github.com/github-copilot/signup), I read this:

`GitHub Copilot is free for maintainers of popular open source software on GitHub`

@wolfgang-ch Do you know if MyTourbook is considered a `popular open source software` and would give access to free GitHub CoPilot ?

**Comment by wolfgang-ch** [comment]
When I try to register

1st Page

![image](https://github.com/mytourbook/mytourbook/assets/1283445/c262225d-a435-45b4-a3dd-7fc6fcf957aa)


2nd Page

![image](https://github.com/mytourbook/mytourbook/assets/1283445/50dccb50-1c7d-476e-b076-91d956808713)


3rd Page 

It is requesting my credit card data but I don't want to give someone my credit card data when I have to pay nothing

![image](https://github.com/mytourbook/mytourbook/assets/1283445/34e35c19-be90-4be4-a997-57c791ac40c2)

**Comment by FJBDev** [comment]
> It is requesting my credit card data but I don't want to give someone my credit card data when I have to pay nothing
> 

Then this probably means that MyTourbook is unfortunately not considered a `popular open source software`  😒  because this is what the [CoPilot FAQ](https://github.com/pricing#faq-copilot) mentions


**A maintainer of a popular open source project is defined as someone who has write or admin access to one or more of the most popular open source projects on GitHub**. Simply visit the GitHub Copilot subscription page to see if you are one of the open source maintainers that meet our criteria for a complimentary subscription. **If you do, you should see that you can add GitHub Copilot for no charge**. If you see a charge on the purchase page then this means that you do not qualify at this time.

**Reply by stvzldn** [reply]
Of course a truly capable AI would be able to apply for a credit card in your name, or at least hallucinate a credit card.

**Comment by FJBDev** [comment]
Considering this, I have reverted to [the TabNine plugin](https://marketplace.eclipse.org/content/tabnine-ai-assistant-chat-software-developers) as they have a Basic subscription that is **free**

So far, it works well in Eclipse

Below is an example of TabNine AI suggestion

![image](https://github.com/mytourbook/mytourbook/assets/25235665/1acf4578-0b8e-4e36-a1c1-f6f79e43b895)


![image](https://github.com/mytourbook/mytourbook/assets/25235665/6d71c605-fc19-4f33-9667-0b0808a64bc2)

**Reply by wolfgang-ch** [reply]
I'll test this AI but after I tested a very fast and promising algorithm which I found yesterday after many hours of research.

Firstly I found the source code https://sourceforge.net/projects/fpf-labeling/files/ which contained the author and then I found the description https://vcg.informatik.uni-rostock.de/~schumann/papers/2008+/particle_based_labeling.pdf


Wikipedia article: https://en.wikipedia.org/wiki/Automatic_label_placement

**Comment by wolfgang-ch** [comment]
@FJBDev 

Is the copyright of the AI generated code suitable with our copyright GNU GPL v2

**Comment by FJBDev** [comment]
> Is the copyright of the AI generated code suitable with our copyright GNU GPL v2

While i am not a lawyer, their [official doc](https://www.tabnine.com/blog/tabnine-enterprise-vs-chatgpt-plus/) seems to say so

>Open source compliance

>The code on which a solution’s AI models are trained can have legal implications for the companies that use the solutions. 

>Tabnine’s AI models are never trained on code with nonpermissive licenses and offer full transparency and attribution.

**Comment by FJBDev** [comment]
GitHub Copilot is now available for Eclipse [here](https://marketplace.eclipse.org/content/github-copilot)
I have tried it and it is pretty good!
